shells/backends/gcp_tf_backend/src/driver.py

```diff
@@ -60,38 +59,39 @@ def get_inventory(self, context):
         resource.add_sub_resource('1', port1)
         return resource.create_autoload_details()
         '''
-        gcp_service = self._can_connect(context)
-        if not gcp_service:
-            raise ValueError("Can't connect to GCP")
+        logger = LoggingSessionContext(context)
+        try:
+            gcp_service = self._can_conntect_to_gcp(context, logger)
+            if not gcp_service:
+                self._raise_and_log(logger, "There was an issue accessing GCP, please check authentication credentials.")  
+            # raise ValueError("Can't connect to GCP")
+        except Exception as e:
+            self._raise_and_log(logger, f"There was an issue initialization GCP provider resource. {e}")    
         return AutoLoadDetails([], [])
         os.remove(DYNAMIC_JSON)
 
     # </editor-fold>
 
     # <editor-fold desc="Validate S3 Bucket Exists">
-    def _validate_bucket_exists(self, bucket_name, context):
-        with LoggingSessionContext(context) as logger:
-            try:
-                storage_client = storage.Client()
-                get_bucket = storage_client.get_bucket(bucket_name)
-                if len(str(get_bucket)) < 0:
-                    raise ValueError('Could not connect: Check credentials')  
-            except Exception as e:
-                self._handle_exception_logging(logger, f"There was an issue accessing the bucket.{e}")        
-
-    def _can_connect(self, context):
-        # api = CloudShellSessionContext(context).get_api()
+    def _validate_bucket_exists(self, bucket_name, context, logger):
```

Comment:
please add type hints - it's a general comment, type hints should be added to all methods